test/test_cuda.py

```diff
@@ -43,6 +44,9 @@
     print('CUDA not available, skipping tests', file=sys.stderr)
     TestCase = object  # noqa: F811
 
+TEST_CUDAMALLOCASYNC = ((os.getenv("PYTORCH_CUDA_ALLOC_CONF") is not None) and
+                        ("backend:cudaMallocAsync" in os.getenv("PYTORCH_CUDA_ALLOC_CONF")))
```

Comment:
This looks bad. We should instead expose a way to query the CUDA allocator configuration from torch itself, so that there is no question whether or not we are in cudaMallocAsync land or not (here, it's possible for the parsing to get desynchronized)

Reply:
agreed, i'll add a python getter.

Reply:
is [this](https://github.com/pytorch/pytorch/pull/65365/commits/3d23053a2924401f1ac33018296b9d4942124283) acceptable? I kinda like having it user-facing and documented. The only thing im not crazy about is hardcoding return strings in `THCPModule_getAllocatorBackend` but that doesn't seem like a big deal.

Reply:
ignore "Pytorch" instead of "PyTorch" again, fixed separately lol